### PR TITLE
GraphNG: add new alpha XY Chart 

### PR DESCRIPTION
**Pull request by ryantxu**
Replacing #29288

The XY chart will have first class support for things where X != time

image

By seperating `Timeseries` from `XY Chart` we can explore how to best support this while keeping the timeseries chart focused on out 95% use case

Unlike the `timeseries` panel, I think this should only work with a single frame, and suggest flattening it explicitly in a transformation

There is a lot more to do to make this really ready for prime time, but I think it is worth adding soon so that it keeps the GraphNG component honest about supporting `x != time`.  This also points out the need to have hints and suggestions for this and the timeseries panel.  In particular:
* warning when x axis is not sorted
* when time exists, but not two numbers, we should suggest "view as time series" or "view as table"
* The legend does not seem to keep up with changes to the structure

### CODE REVIEW

File: public/app/plugins/panel/xychart/module.tsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+      [FieldConfigProperty.Color]: {
+        settings: {
+          byValueSupport: false,
+        },
+        defaultValue: {
+          mode: FieldColorModeId.PaletteClassic,
+        },
+      },
+    },
+    useCustomConfig: builder => {
```
Comment (torkelo) [inline public/app/plugins/panel/xychart/module.tsx:36]: would be nice to extract this and share the common definitions

Comment (ryantxu) [reply public/app/plugins/panel/xychart/module.tsx]: done ;)

File: packages/grafana-ui/src/components/uPlot/config.ts [hunk 2]
```diff
@@ -70,16 +70,16 @@ export interface LineConfig {
 /**
  * @alpha
  */
-export interface AreaConfig {
+export interface FillConfig {
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/uPlot/config.ts:73]: This change could be its own PR... it touches a bunch of files, but the shared config felt awkward with "Area" rather than "Fill"

File: packages/grafana-data/src/types/plugin.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { LiveChannelSupport } from './live';
 export enum PluginState {
   alpha = 'alpha', // Only included if `enable_alpha` config option is true
   beta = 'beta', // Will show a warning banner
-  stable = 'stable', // Will not show anything 
+  stable = 'stable', // Will not show anything
```
Comment (ryantxu) [inline packages/grafana-data/src/types/plugin.ts:9]: build fails without this change (lint)

**Review by torkelo — APPROVED** [review]
Think this looks mergable, and see that you fixed a master lint error as well
